### DISCUSSION TITLE
Create and give role persissions to user in Unit testing

### DISCUSSION THREAD

**Discussion by Antoine-lb** [strapi/strapi#7193] [Legacy Topics]
Hello,
I followed the [unit test doc](https://strapi.io/documentation/v3.x/guides/unit-testing.html#testing-auth-endpoint-controller) and everything works.

But I want to test and enpoint on `create`, and for that I need the user to be allowed to create to a specific field.

I was trying to play directly with the Strapi functions but I didn't managed to make it work.
Any help would be usefull, Thanks.

**Comment by lauriejim** [comment]
Hello! I suggest you to use 
```js
const permission = await strapi.query('permission', 'users-permissions').findOne({
  type: 'application',
  controller: 'yourControllerName',
  action: 'create',
 role: theIdOfTheRole(Int)
});

permission.enabled = 1;

await strapi.query('permission', 'users-permissions').update({id: permission.id}, permission);
```
something like this.

**Comment by Antoine-lb** [comment]
Roles for Authenticated is always 1 and public always 2 right? or it can change depending on the boot-up of the server?
Thanks!

**Reply by derrickmehaffy** [reply]
Depends on the database but generally yes, you can always query the roles to confirm
